Fix status label

StefRe opened the pull request:
- Fix status_label if no results to show
   status_label update is triggered by assignment of test results in
    process_finished. In case of no results, process_finished is called
    with an empty list of testresults. This empty list must also be
    assigned in order to trigger the status_label update, otherwise the
    status remains "Running tests ..." after finishing. So we have to
    differentiate between None and [] in the if clause.

    Without the fix it looks like this after running when no tests were discovered:
    ![grafik](https://user-images.githubusercontent.com/19879328/78247863-e963f480-74eb-11ea-9399-e7d70ec511b2.png)

- Reset status_label when running tests is stopped
    and make parsing errors in unittestrunner.load_data indeed ignored silently
    (it relies on the condition that the result includes the final line of all
    dashes, otherwise an Indexerror will occur)

    Without the fix the status label "Running tests ..." remains visible after stopping
    the run.

jitseniesen commented:
>     * Fix status_label if no results to show

When I tried your PR, I didn't see any change here. If I use the plugin to run pytest in a directory with no tests, the label still shows "Running tests ..." after pytest is finished.

Further testing suggests this depends on the testing framework. If I use the unittest framework, then your fix does work correctly.

Could you please have a look? Which testing framework did you use? The way the pytest framework is handled in the plugin is a bit more complicated, so there is more scope for errors.

jitseniesen commented:
I forgot to say, thanks for your work and especially for including the tests!

StefRe commented:
Oh - this is the problem when we don't have enough tests :). At first I wanted to add an integration test but then decided for just the unit test (`test_unittestwidget_process_finished_updates_status_label`).  

So anyway I fixed it now and added a small integration test (`test_run_with_no_tests_discovered_and_display_results`) for all three test frameworks. Thanks for pointing it out to me.

jitseniesen commented on spyder_unittest/widgets/tests/test_unittestgui.py:
```diff
@@ -224,6 +224,28 @@ def test_run_tests_using_unittest_and_display_results(qtbot, tmpdir,
+                                                          monkeypatch, framework):
+    """Basic integration test."""
+    os.chdir(tmpdir.strpath)
+    testfilename = tmpdir.join('test_foo.py').strpath
```
I think you can remove this line.

StefRe replied:
Yes, you're right. I overlooked this one. Thanks.

jitseniesen commented:
Looks good now, this is going in. Thanks!
